portmap/core/views.py

```diff
@@ -169,35 +175,56 @@ def find_articles(request):
         form = QueryIndexForm(data=None, datatypes=datatypes)
         return TemplateResponse(request, "core/index.html", {'form': form, 'datatypes': datatypes})
 
+# Core related functions
+def index(request):
+    return TemplateResponse(request, "core/index.html", _get_index_context())
 
-@ux_requires_post
-def article_feedback(request, article_name):
-    form = ArticleFeedbackForm(data=request.POST)
-    Feedback.objects.create(article=Article.objects.get(name=article_name),
-                            reaction=form.data['reaction'],
-                            explanation=form.data['explanation'])
-    message = "*New article feedback for \"" + article_name + "\"*:\n\nReaction: " + form.data['reaction'] + "\n\n" + form.data['explanation']
-    notify(message)
-    return TemplateResponse(request, "core/thankyou.html")
+def about(request):
+    return TemplateResponse(request, "core/about.html")
 
+def _get_index_context():
+    query_structure = Article.get_query_structure()
+    query_form = QueryIndexForm(data=None, datatypes=query_structure.keys())
+    feedback_form = UseCaseFeedbackForm(data=None, datatype='None', source='', destination='')
+    datatype_help = GithubClient().get_datatype_help()
+    datatype_help_cleaned = {datatype: f"{datatype_help.get(datatype, '')}" for datatype in query_structure.keys()}
 
-@ux_requires_post
-def usecase_feedback(request):
-    feedback = UseCaseFeedbackForm(data=request.POST);
-    if feedback.is_valid():
-        feedback.save()
-        message = "*New use case feedback:*\n\n" + feedback.data['explanation']
-        notify(message)
 
-    return TemplateResponse(request, "core/thankyou.html")
+    return {'form': query_form,
+               'query_structure': json.dumps(query_structure),
+               'use_case_form': feedback_form,
+               'datatypes': query_structure.keys(),
+               'datatype_help': datatype_help_cleaned}
 
+def login_as_user(request):
```

Comment:
Try not to refactor files to move things around at the same PR as modifying logic - it makes it harder to review the logic.  That said I'm very much in favour of reorganizing code to be more readable and we don't need an issue open to make a PR that reorganizes the order of all the methods of a file.

Reply:
Got it! I'll make a different PR with reorg some of the methods and classes to make it more readable!